packages/devtools-utils/src/worker-utils.js

```diff
@@ -34,55 +34,83 @@ WorkerDispatcher.prototype = {
+        }
 
-        const listener = ({ data: result }) => {
-          if (result.id !== id) {
-            return;
-          }
+        calls.push([args, resolve, reject]);
 
-          if (!this.worker) {
-            return;
-          }
+        if (!queue) {
+          flush();
+        }
+      });
+    }
+
+    const flush = () => {
+      const items = calls.slice();
+      calls.length = 0;
+
+      const id = this.msgId++;
+      this.worker.postMessage({ id, method, calls: items.map(item => item[0]) });
+
+      const listener = ({ data: result }) => {
+        if (result.id !== id) {
+          return;
+        }
+
+        if (!this.worker) {
+          return;
+        }
+
+        this.worker.removeEventListener("message", listener);
 
-          this.worker.removeEventListener("message", listener);
-          if (result.error) {
-            reject(result.error);
+        result.results.forEach((resultData, i) => {
+          const [, resolve, reject] = items[i];
+
+          if (resultData.error) {
+            reject(resultData.error);
           } else {
-            resolve(result.response);
+            resolve(resultData.response);
           }
-        };
+        });
+      };
 
-        this.worker.addEventListener("message", listener);
-      });
-    };
+      this.worker.addEventListener("message", listener);
+    }
+
+    return (...args: any) => push(args);
   }
 };
 
 function workerHandler(publicInterface: Object) {
   return function (msg: Message) {
-    const { id, method, args } = msg.data;
-    try {
-      const response = publicInterface[method].apply(undefined, args);
-      if (response instanceof Promise) {
-        response.then(
-          val => self.postMessage({ id, response: val }),
-          // Error can't be sent via postMessage, so be sure to
-          // convert to string.
-          err => self.postMessage({ id, error: err.toString() })
-        );
-      } else {
-        self.postMessage({ id, response });
+    const { id, method, calls } = (msg.data: any);
+
+    Promise.all(calls.map(args => {
+      try {
+        const response = publicInterface[method].apply(undefined, args);
+        if (response instanceof Promise) {
+          return response.then(
+            val => ({ response: val }),
+            // Error can't be sent via postMessage, so be sure to
+            // convert to string.
+            err => ({ error: err.toString() })
+          );
+        } else {
+          return { response };
+        }
+      } catch (error) {
+        // Error can't be sent via postMessage, so be sure to convert to
+        // string.
+        return { error: error.toString() };
       }
-    } catch (error) {
-      // Error can't be sent via postMessage, so be sure to convert to
-      // string.
-      self.postMessage({ id, error: error.toString() });
-    }
+    })).then(results => {
+      self.postMessage({ id, results });
+    });
   };
 }
 
```

Comment:
None of this updates `streamingWorkerHandler` below here because as far as I can tell it is already incompatible with the existing `.task` function in the general case. `.task` only listens for a single message, so it can't be used alongside `streamingWorkerHandler` as is.